statsmodels/tsa/statespace/innovation.py

```diff
@@ -0,0 +1,72 @@
+from statsmodels.tsa.statespace.mlemodel import MLEModel, MLEResults, MLEResultsWrapper
+
+
+class InnnovationModel(MLEModel):
+    def __init__(self, endog, k_states, **kwargs):
+        kwargs.update(k_posdef=1)
+        k_states += 1
+        super().__init__(endog, k_states, **kwargs)
+        self.ssm['design', 0, -1] = 1
+        self.ssm['selection', -1, 0] = 1
+
+    def __setitem__(self, key, value):
+
+        _type = type(key)
+        if _type is str:
+            if key == 'cov':
+                self.ssm['state_cov', 0, 0] = value
+            elif key == 'transition':
+                self.ssm['transition', :-1, :-1] = value
+            elif key == 'selection':
+                self.ssm['transition', :-1, -1:] = value
+            elif key == 'design':
+                self.ssm['design', :, :-1] = value
+            elif key in ('state_cov', 'obs_cov'):
+                self.ssm['state_cov', :, :] = value
+            else:
+                return super().__setitem__(key, value)
+        elif _type is tuple:
+            name, slice_ = key[0], key[1:]
+            if name == 'transition':
+                self.ssm['transition', :-1, :-1][slice_] = value
```

Comment:
I think this method of setting these matrices unfortunately isn't going to work because it first gets the matrix and then sets values in the slice. The problem is that if, for example, `value` has complex values and `self.ssm['transition', :-1, :-1]` is a real matrix, then it will just truncate the complex part and issue a warning.